WordPress/Classes/ViewRelated/Site Creation/Web Address/DesignSystem/Components/DSButton.swift

```diff
@@ -0,0 +1,165 @@
+import SwiftUI
+
+public struct DSButton: View {
```

Comment:
This is a new custom view rather than a modifier on `SwiftUI.Button` for two reasons:
1. It contains things like `ProgressView` that automatically replaces its text when the state is `isLoading` equals true.
2. It would be too intrusive as a modifier. SwiftUI modifiers do one thing on the view. This would have affects that isn't perhaps expected.

Reply:
I think that could be fine. However at any place both libraries are imported, people would need to specify in either case right? I don't feel strongly about it. The `DS` prefix doesn't look great imo too. I think it might be confusing for people. Sometimes Xcode raises strange errors too on such issues that doesn't reveal the problem really. I think we can discuss on (hopefully with others joining the chat) and prepare another PR with all the other changes. You have also mentioned that you have architectural ideas around all this.